src/cc/tests/graph_test.cc

```diff
@@ -1004,5 +1004,177 @@ TEST(GraphTest, GetNodeTypesAcrossPartitions)
     EXPECT_EQ(std::vector<snark::Type>({0, 2, 1, 2, 2, -1}), types);
 }
 
+// Neighbor Count Tests
+TEST(GraphTest, GetNeigborCountSinglePartition)
+{
+    TestGraph::MemoryGraph m1;
+
+    /*
+
+    Insert first node and neighbors
+
+    (0, 0, 1.0)
+    |         |
+    v         v
+    (3,0,1.0)  (4,0,1.0)
+
+    */
+
+    m1.m_nodes.push_back(
+        TestGraph::Node{.m_id = 0,
+                        .m_type = 0,
+                        .m_weight = 1.0f,
+                        .m_neighbors{std::vector<TestGraph::NeighborRecord>{{1, 0, 1.0f}, {2, 0, 1.0f}}}});
+
+    /*
+
+    Insert second node and neighbors
+
+    (1, 1, 1.0) -----> (5, 1, 1.0f)
+    |         |
+    v         v
+    (3,0,1.0)  (4,0,1.0)
+
+    */
+
+    m1.m_nodes.push_back(TestGraph::Node{
+        .m_id = 1,
+        .m_type = 1,
+        .m_weight = 1.0f,
+        .m_neighbors{std::vector<TestGraph::NeighborRecord>{{3, 0, 1.0f}, {4, 0, 1.0f}, {5, 1, 1.0f}}}});
+
+    // Initialize graph
+    auto path = std::filesystem::temp_directory_path();
+    TestGraph::convert(path, "0_0", std::move(m1), 2);
+    snark::Graph g(path.string(), {0}, snark::PartitionStorageType::memory, "");
+
+    // Check for singe edge type filter
+    std::vector<snark::NodeId> nodes = {0, 1};
+    std::vector<snark::Type> types = {0};
+    std::vector<uint64_t> output_neighbors_count(nodes.size());
+
+    g.NeighborCount(std::span(nodes), std::span(types), output_neighbors_count);
+    EXPECT_EQ(std::vector<uint64_t>({2, 2}), output_neighbors_count);
+
+    // Check for different singe edge type filter
+    types = {1};
+    std::fill_n(output_neighbors_count.begin(), 2, 0);
+
+    g.NeighborCount(std::span(nodes), std::span(types), output_neighbors_count);
+    EXPECT_EQ(std::vector<uint64_t>({0, 1}), output_neighbors_count);
+
+    // Check for both edge types
+    types = {0, 1};
+    std::fill_n(output_neighbors_count.begin(), 2, 0);
+
+    g.NeighborCount(std::span(nodes), std::span(types), output_neighbors_count);
+    EXPECT_EQ(std::vector<uint64_t>({2, 3}), output_neighbors_count);
+
+    // Check returns 0 for unsatisfying edge types
+    types = {-1, 100};
+    std::fill_n(output_neighbors_count.begin(), 2, 0);
```

Comment:
fill array with -1s to verify it was actually updated, because it can be not touched at all.